tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -1054,6 +1056,37 @@ def test_group_experimental_warning() -> None:
         _ = TPESampler(multivariate=True, group=True)
 
 
+def run_tpe(k: int, sequence_dict: Dict[int, List[int]], hash_dict: Dict[int, int]) -> None:
+    hash_dict[k] = hash("nondeterministic hash")
+    sampler = TPESampler(n_startup_trials=1, seed=2, multivariate=True, group=True)
+    study = create_study(sampler=sampler)
+    sequence = []
+    for _ in range(10):
+        trial = study.ask()
+        picked = [i for i in range(10) if trial.suggest_int(str(i), 0, 1) == 1]
+        study.tell(trial, len(picked))
+        sequence.extend(picked)
+    sequence_dict[k] = sequence
```

Comment:
How about simplifying the test using a standard `Study.optimize`? For example, I confirmed that the following code pass with this PR, but fails with the master.
```suggestion
    sampler = TPESampler(n_startup_trials=1, seed=2, multivariate=True, group=True)
    study = create_study(sampler=sampler)
    study.optimize(lambda t: np.sum(t.suggest_int(f"x{i}", 0, 10) for i in range(10)), n_trials=2)
    sequence_dict[k] = list(study.trials[-1].params.values())
```